src/box/iproto_features.h

```diff
@@ -0,0 +1,118 @@
+/*
+ * SPDX-License-Identifier: BSD-2-Clause
+ *
+ * Copyright 2010-2021, Tarantool AUTHORS, please see AUTHORS file.
+ */
+#pragma once
+
+#include <assert.h>
+#include <stdint.h>
+#include <string.h>
+
+#include "bit/bit.h"
+
+#if defined(__cplusplus)
+extern "C" {
+#endif
+
+/**
+ * IPROTO protocol feature ids returned by the IPROTO_ID command.
+ */
+enum iproto_feature_id {
+	/**
+	 * Streams support: IPROTO_STREAM_ID header key.
+	 */
+	IPROTO_FEATURE_STREAMS = 0,
+	/**
+	 * Transactions in the protocol:
+	 * IPROTO_BEGIN, IPROTO_COMMIT, IPROTO_ROLLBACK commands.
+	 */
+	IPROTO_FEATURE_TRANSACTIONS = 1,
+	iproto_feature_id_MAX,
+};
+
+/**
+ * IPROTO protocol feature bit map.
+ */
+struct iproto_features {
+	char bits[BITMAP_SIZE(iproto_feature_id_MAX)];
```

Comment:
IMO, would be fine to hardcode the size here, and add an assertion to `iproto_features_init` that `bitmap_size()` is the same as the hardcoded value. Then wouldn't need the first commit. But I am ok with this solution too.

Reply:
Using bitmap_size at compile time is more convenient IMO. Let's leave it as is.